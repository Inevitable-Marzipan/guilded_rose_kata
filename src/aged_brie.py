from src.gilded_item import GildedItem

class AgedBrie(GildedItem):
    QUALITY_MODIFIER = 1