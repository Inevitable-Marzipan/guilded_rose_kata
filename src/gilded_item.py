from src.item import Item

class GildedItem(Item):
    QUALITY_MODIFIER = -1

    def change_sell_in(self):
        self.sell_in -= 1
    
    def change_quality(self):
        if (self.quality < 50) and ((self.quality + self.QUALITY_MODIFIER) >= 0):
            self.quality += self.QUALITY_MODIFIER

    def update_item(self):

        self.change_quality()

        self.change_sell_in()

        if self.sell_in < 0:
            self.change_quality()
