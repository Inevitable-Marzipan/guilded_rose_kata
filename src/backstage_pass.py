from src.gilded_item import GildedItem

class BackstagePass(GildedItem):
    QUALITY_MODIFIER = 1

    def update_item(self):
        print(self)
        self.change_quality()

        if self.sell_in < 11:
            self.change_quality()
        if self.sell_in < 6:
            self.change_quality()

        self.change_sell_in()

        if self.sell_in < 0:
            self.quality = 0