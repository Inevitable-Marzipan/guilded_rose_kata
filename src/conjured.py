from src.gilded_item import GildedItem

class Conjured(GildedItem):
    QUALITY_MODIFIER = -2