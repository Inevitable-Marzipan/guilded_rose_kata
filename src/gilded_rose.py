# -*- coding: utf-8 -*-
from src.gilded_item import GildedItem
from src.aged_brie import AgedBrie
class GildedRose(object):

    def __init__(self, items):
        self.items = items

    def update_quality(self):
        for item in self.items:
            item.update_item()


