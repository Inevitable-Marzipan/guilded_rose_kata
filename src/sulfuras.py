from src.gilded_item import GildedItem

class Sulfuras(GildedItem):
    
    def change_quality(self):
        pass

    def change_sell_in(self):
        pass