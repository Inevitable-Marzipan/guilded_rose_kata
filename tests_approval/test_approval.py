import pytest
from approvaltests.approvals import verify

from src.gilded_rose import Item, GildedRose


def gilded_rose_simulation(items, days):
    result = ""
    for day in range(days):
        result += f"-------- day {day} -------- \n"
        result += "name, sellIn, quality \n"
        for item in items:
            result += str(item)
            result += "\n"
        result += "\n"
        GildedRose(items).update_quality()
    return result

def test_items():
    items = [
            Item(name="Aged Brie", sell_in=2, quality=0),
        ]
    days = 5
    verify(gilded_rose_simulation(items, days))

def test_items_high_quality():
    items = [
            Item(name="Aged Brie", sell_in=10, quality=49),
            Item(name="Backstage passes to a TAFKAL80ETC concert", sell_in=10, quality=49),
            Item(name="Sulfuras, Hand of Ragnaros", sell_in=10, quality=49),
            Item(name="Item of Randomness", sell_in=10, quality=49)
        ]
    days = 3
    verify(gilded_rose_simulation(items, days))
    

def test_items_negative_sellin_high_quality():
    items = [
            Item(name="Aged Brie", sell_in=1, quality=49),
            Item(name="Backstage passes to a TAFKAL80ETC concert", sell_in=1, quality=49),
            Item(name="Sulfuras, Hand of Ragnaros", sell_in=1, quality=49),
            Item(name="Item of Randomness", sell_in=1, quality=49)
        ]
    days = 3
    verify(gilded_rose_simulation(items, days))

def test_adhoc_item_combos():
    items = [
             Item(name="Backstage passes to a TAFKAL80ETC concert", sell_in=12, quality=30),
             Item(name="Backstage passes to a TAFKAL80ETC concert", sell_in=7, quality=30),
             Item(name="Backstage passes to a TAFKAL80ETC concert", sell_in=5, quality=30),
             Item(name="Item of Randomness", sell_in=1, quality=0),
             Item(name="Item of Randomness", sell_in=1, quality=-1),
             Item(name="An item of Randomness", sell_in=1, quality=-1),
             Item(name="Aged Brie", sell_in=-1, quality=51)
    ]
    days = 3
    verify(gilded_rose_simulation(items, days))