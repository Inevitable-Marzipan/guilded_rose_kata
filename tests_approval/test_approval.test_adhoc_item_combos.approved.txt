-------- day 0 -------- 
name, sellIn, quality 
Backstage passes to a TAFKAL80ETC concert, 12, 30
Backstage passes to a TAFKAL80ETC concert, 7, 30
Item of Randomness, 1, 0

-------- day 1 -------- 
name, sellIn, quality 
Backstage passes to a TAFKAL80ETC concert, 11, 31
Backstage passes to a TAFKAL80ETC concert, 6, 32
Item of Randomness, 0, 0

-------- day 2 -------- 
name, sellIn, quality 
Backstage passes to a TAFKAL80ETC concert, 10, 32
Backstage passes to a TAFKAL80ETC concert, 5, 34
Item of Randomness, -1, 0

